### PR TITLE
require writing test case for verification suites

### PR DESCRIPTION
**Pull request by jbwheatley**

### CODE REVIEW

**Comment by jbwheatley** [comment]
ugh, this message pact stuff.. the message pact suites are inconsistently failing, saying `Verification Failed - Uncaught exception during scan` and seeing a ton of stuff like `[25.046s][warning][gc,alloc] ClassGraph-worker-18: Retried waiting for GCLocker too often allocating 256 words` 🤦 
@kflorence FYI

**Comment by solarmosaic-kflorence** [comment]
@jbwheatley oh nice, that's happening in `main` too? 😅  see #20

**Comment by jbwheatley** [comment]
I don't really understand why the changes in this PR cause the GC error when the test suites all run on `main` 🤷‍♂️

**Comment by solarmosaic-kflorence** [comment]
@jbwheatley I think it's a bit flaky, I wouldn't be surprised if it were happening also in `main`. I think the classpath scanning stores a lot of stuff in memory, so the fix for now might be just giving the build more memory. We do have a slightly odd case here, where we have many suites that all run tests at the same time and require classpath scanning -- in a normal project, there would only be one suite running provider tests -- although, since the scanning happens for every consumer I'd imagine this problem would also occur for a single provider with many consumers.

**Comment by jbwheatley** [comment]
I turned off parallel execution in the tests and they all passed haha. Seems like you had the right idea about why the issue was happening!

**Comment by solarmosaic-kflorence** [comment]
@jbwheatley yeah, I tried that too, and it did help, although when I ran the tests multiple times it was still a bit flaky (at least in my new PR). I think it might still need a bump in memory (although perhaps not as high as 4GB).

File: shared/src/main/scala/pact4s/PactVerifyResources.scala [hunk 1]
```diff
@@ -25,9 +28,20 @@ trait PactVerifyResources {
 
   private[pact4s] val verifier = new ProviderVerifier()
 
-  private[pact4s] def verifySingleConsumer(consumer: IConsumerInfo): Unit
+  private[pact4s] def failure(message: String)(implicit fileName: FileName, file: File, line: Line): Nothing
+
+  private[pact4s] def verifySingleConsumer(
```
Comment (solarmosaic-kflorence) [inline shared/src/main/scala/pact4s/PactVerifyResources.scala:33]: 👍  i was wondering if we should consolidate this into one place as well.

File: shared/src/main/scala/pact4s/PactVerifyResources.scala [hunk 2]
```diff
@@ -25,9 +28,20 @@ trait PactVerifyResources {
...
+      consumer: IConsumerInfo
+  )(implicit fileName: FileName, file: File, line: Line): Unit =
+    verifier.runVerificationForConsumer(new java.util.HashMap[String, Object](), providerInfo, consumer) match {
+      case failed: VerificationResult.Failed =>
+        verifier.displayFailures(List(failed).asJava)
+        failure(s"Verification failed:\n ${failed.toString}")
```
Comment (kflorence) [inline shared/src/main/scala/pact4s/PactVerifyResources.scala:39]: does this help with the error reporting?

Comment (jbwheatley) [reply shared/src/main/scala/pact4s/PactVerifyResources.scala:39]: yes, this should put the entire failure output from the verifier into the place where test failures are displayed in scalatest etc. Might actually remove the need for stacktrace to be turned on on the verifier. We can come back and review it later if it turns out we are being far too verbose in the error outputs

Comment (jbwheatley) [reply shared/src/main/scala/pact4s/PactVerifyResources.scala:39]: actually, the stack-trace is really noisy, and this output gives a good amount of. info, so I'm turning stack-trace printing off by default. Can still be turned on though

Comment (solarmosaic-kflorence) [reply shared/src/main/scala/pact4s/PactVerifyResources.scala:39]: cool, i will test this out with the reproducer for issue 19 -- that is where i tested out the stacktrace as well

**Pull request merged**